README.md

```diff
@@ -2,30 +2,384 @@
+into the first call to `forEach`, while a method reference is passed
+into the second call.
+
+```java
+import java.util.List;
+
+public class Example {
+    public static void main(String[] args) {
+        List<Integer> ages = List.of(12, 55, 37, 9);
+        
+        System.out.println("Pass lambda expression as consumer");
+        ages.stream().forEach(s -> System.out.println("next number is " + s));
+        
+        System.out.println("Pass method reference as consumer");
+        ages.stream().forEach(System.out::println);
+    }
+}
+```
+
+The program prints:
+
+```text
+Pass lambda expression as consumer
+next number is 12
+next number is 55
+next number is 37
+next number is 9
+Pass method reference as consumer
+12
+55
+37
+9
+```
+
+## allMatch, anyMatch, noneMatch
+
+`boolean	allMatch(Predicate<? super T> predicate)`  
+`boolean	anyMatch(Predicate<? super T> predicate)`  
+`boolean	noneMatch(Predicate<? super T> predicate)`
+
+- The `allMatch` method takes a `Predicate` parameter and returns a `boolean` 
+  indicating whether all elements match the predicate.
+- The `anyMatch` method takes a `Predicate` parameter and returns a `boolean`
+  indicating whether at least one element matches the predicate.
+- The `noneMatch` method takes a `Predicate` parameter and returns a `boolean`
+  indicating whether no elements match the predicate.
+
+```java
+import java.util.function.Predicate;
+
+public class Example {
+    public static void main(String[] args) {
+        List<Integer> ages = List.of(22, 55, 37, 19);
+        Predicate<Integer> atleast18 = i -> i >= 18;
+        Predicate<Integer> allOdd = i -> i % 2 == 1;
+        Predicate<Integer> over60 = i -> i > 60;
+
+        System.out.println("All at least 18? " + ages.stream().allMatch(atleast18));  //true
+        System.out.println("All odd? " + ages.stream().allMatch(allOdd)); //false
+        System.out.println("Any odd? " + ages.stream().anyMatch(allOdd)); //true
+        System.out.println("None over 60? " + ages.stream().noneMatch(over60)); //true
+    }
+}
+```
+
+The program prints:
+
+```text
+All at least 18? true
+All odd? false
+Any odd? true
+None over 60? true
+```
+
+### matching with primitive data types
+
+Recall that we create primitive streams `IntStream`, `DoubleStream`, and `LongStream`
+when the elements are primitive values (int, double, long).  
+
+A primitive predicate `IntPredicate`, `DoublePredicate`, and `LongPredicate`
+is required to match primitive stream elements.
+
+```java
+import java.util.Arrays;
+import java.util.function.IntPredicate;
+import java.util.stream.IntStream;
+
+public class ExampleIntPredicateMatch {
+    public static void main(String[] args) {
+        int[] grades = {90, 75, 100, 58};
+        IntPredicate isPassing = i -> i >= 60;
+        IntStream gradesStream = Arrays.stream(grades);
+        System.out.println("All at least 18? " + gradesStream.allMatch(isPassing));  //false
```

Comment:
I think "All at least 18?" should really be "All passing?" (or something to that affect)

Reply:
thanks! I fixed the print statement